File: examples/event/subscribe/node/subscribeToATopic.js
const { EventService } = require("m3o/event");

// Subscribe to messages for a given topic.
async function subscribeToAtopic() {
  let eventService = new EventService(process.env.MICRO_API_TOKEN);
  let rsp = await eventService.subscribe({
    topic: "user",
  });
  console.log(rsp);
}

subscribeToAtopic();

File: examples/event/subscribe/go/subscribeToATopic.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/event"
)

// Subscribe to messages for a given topic.
func SubscribeToAtopic() {
	eventService := event.NewEventService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := eventService.Subscribe(&event.SubscribeRequest{
		Topic: "user",
	})
	fmt.Println(rsp, err)
}
